### PR TITLE
OS detection fixing gutter error notifications

### LINKED ISSUE #29: Fix PythonBuddy not working in Windows
Line Numbers do not show up correctly when PythonBuddy is running on a Windows system. 

Thank you @DirtySoc for raising this up and providing these screenshots:
Windows 10 1803 w/ Python3.7:
47252013-2547ee80-d40b-11e8-9fcc-6ebad2e17c44

Ubuntu 18.01 w/ Python3.6.6:
47252254-a99c7080-d40f-11e8-93f7-7f07dbbbf819

PythonBuddy was developed for Unix systems so it works on Mac and Ubuntu for example. Unfortunately, the line numbers are not processed correctly in PythonBuddy for Windows. My hunch is that in app.py the data passed in the AJAX calls is formatted differently in Windows systems. Or, there is an issue with creating temp files in Windows system.

### PR DESCRIPTION
**Pull request by DirtySoc**
tested w/ python 3.7.0, windows 10 1809, and MS Edge.

image

Fixes #29

### CODE REVIEW

**Comment by ethanchewy** [comment]
@DirtySoc 

For scoping and to create clearer error messages, let's do something like this:
```
line_num = None
if os.name == "nt":
        line_num = error.split(":")[2]
else:
       try:
            line_num = error.split(":")[1]
      except Exception as e:
           print(os.name + " not compatible: " + e)
```

File: app.py [hunk 1]
```diff
@@ -161,7 +161,11 @@ def process_error(error):
     if error.find("Your code has been rated at") > -1:
         return None
 
-    line_num = error.split(":")[1]
+    # Detect OS
+    if os.name == "nt":
+        line_num = error.split(":")[2]
+    else:
+        line_num = error.split(":")[1]
```
Comment (ethanchewy) [inline app.py]: Could you add a try and except block here just in case another operating system is not compatible?

**Comment by DirtySoc** [comment]
updated. let me know if that works!

File: app.py [hunk 2]
```diff
@@ -161,7 +161,11 @@ def process_error(error):
     if error.find("Your code has been rated at") > -1:
         return None
 
-    line_num = error.split(":")[1]
+    # Detect OS
```
[the remark this answers is not in the document]
Comment (DirtySoc) [reply app.py]: Do you mean above the comment? Not sure where "over here is: 
image

**Pull request merged**